### ISSUE TITLE
fix testFileLoggingMaxSize unit test

### ISSUE THREAD

**Issue by jayahariv** [couchbase/couchbase-lite-android#1835]
Referencing the parent task: #1808

[Link to Test](https://github.com/couchbase/couchbase-lite-android/blob/master/android/CouchbaseLite/src/androidTest/java/com/couchbase/lite/LogTest.java#L263)

conversations happened so far: 

@jayahariv 22 hours ago  Member
ignored since the debug logs are not printing anything in it

  @borrrden
borrrden 16 hours ago  Member
So why ignore instead of adjusting the test? For release mode it's true no debug logs should be written but in debug mode the debug logs should be written.

  @jayahariv
jayahariv 16 hours ago  Member
waiting to discuss with Pasin/Sandy tomorrow, and need to understand how the debug and release works in Android.

  @pasin
pasin 14 hours ago  Member
Android Studio has a separate option/button for run (release build type) and debug. To enable debug build type, select debug.
I understand that when running test from Jenkins using gradle command, the release build type will be used.
As @borrrden suggested, I think we should modify the test to ignore or not checking debug.

@jayahariv
jayahariv 5 hours ago  Member
I used below code to check whether the test is in debug or not. and I am getting isDebug=true but still, cbl_debug_xxx file only with heading.

any suggestions
```
boolean isDebug = (
                        (context.getApplicationInfo().flags & ApplicationInfo.FLAG_DEBUGGABLE) != 0
```
  @pasin
pasin 34 minutes ago  Member
I guessed CMAKE_BUILD_TYPE doesn't get set correctly according to the android build type.

We may need to explicitly set the CMAKE_BUILD_TYPE to the right value in https://github.com/couchbase/couchbase-lite-android/blob/master/android/CouchbaseLite/build.gradle#L86-L97.

**Comment by bmeike** [comment]
`isDebuggable` is not what you want.  You might want `BuildConfig.DEBUG`, but that tells whether the application was built debug or not.  If you want to know whether an included library was built debug, then you probably want something like this in the build for the library:
`buildTypes {
        debug {
            buildConfigField "Boolean", "CB_DEBUG", "true"
        }
        
        ...
    }`

**Comment by jayahariv** [comment]
after spending lots of time, we figured the log.d was not called and implemented. So adding the fix as well as enabling the test in PR #1837.

**Comment by pasin** [comment]
After all, it seems like gradle does set CMAKE_BUILD_TYPE correctly based on the build type.
